source/FreeRTOS_UDP_IP.c

```diff
@@ -288,14 +281,13 @@ BaseType_t xProcessReceivedUDPPacket( NetworkBufferDescriptor_t * pxNetworkBuffe
 
     if( pxUDPPacket->xEthernetHeader.usFrameType == ipIPv4_FRAME_TYPE )
     {
-        xReturn = xProcessReceivedUDPPacket_IPv4( pxNetworkBuffer,
-                                                  usPort, pxIsWaitingForARPResolution );
+        xProcessReceivedUDPPacket_IPv4( pxNetworkBuffer,
+                                        usPort, pxIsWaitingForARPResolution );
     }
-
-    if( pxUDPPacket->xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE )
+    else if( pxUDPPacket->xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE )
     {
-        xReturn = xProcessReceivedUDPPacket_IPv6( pxNetworkBuffer,
-                                                  usPort, pxIsWaitingForARPResolution );
+        xProcessReceivedUDPPacket_IPv6( pxNetworkBuffer,
```

Comment:
Can we return the return value of the corresponding the IPV4/V6 functions as well to enable testing?

Reply:
All functions that handle incoming packets may either use the network buffer to return it, or return false, meaning : the buffer was not released.
I found that network buffers got lost because a "pdTRUE" was returned.

Reply:
In this case aren't we returning the packets using the network buffers?

The `xReturn`  in this case is set to `pdTRUE` always.